src/resolve/resolvectl.c

```diff
@@ -1828,25 +1827,27 @@ static int verb_dns(int argc, char **argv, void *userdata) {
         _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
         _cleanup_(sd_bus_message_unrefp) sd_bus_message *req = NULL;
         sd_bus *bus = userdata;
-        int ifindex, r;
         char **p;
+        int r;
 
         assert(bus);
 
-        if (argc <= 1)
-                return status_all(bus, STATUS_DNS);
-
-        ifindex = parse_ifindex_with_warn(argv[1]);
-        if (ifindex < 0)
-                return ifindex;
-
-        if (ifindex == LOOPBACK_IFINDEX) {
+        if (arg_ifindex == LOOPBACK_IFINDEX) {
                 log_error("Interface can't be the loopback interface (lo). Sorry.");
                 return -EINVAL;
         }
 
-        if (argc == 2)
-                return status_ifindex(bus, ifindex, NULL, STATUS_DNS, NULL);
+        if (argc <= 1) {
+                if (arg_ifindex <= 0)
+                        return status_all(bus, STATUS_DNS);
+                else
+                        return status_ifindex(bus, arg_ifindex, NULL, STATUS_DNS, NULL);
```

Comment:
This changes
```console
$ resolvectl dns eth0
```
to
```console
$ resolvectl dns --interface=eth0
```
I'm not convinced that this is better. The syntax with link as a positional argument is very concise and feels pretty natural.

Maybe this would be easier to understand if you gave some example of future extension that you have in mind.